### PR TITLE
Rewrite az in python

### PR DESCRIPTION
**Pull request by suominen**
Use python instead of bash, so that we get benefits from using `#!/usr/bin/env python`:

* `distutils` will replace it with an absolute path to the specific python interpreter being used
  * helps with `pip install` (#1043)
  * helps with tools like `pipx` (pipxproject/pipx#162)
* if running inside a `virtualenv` as is, it will run the python interpreter for that `virtualenv` (#3989)

There doesn't seem to be a measurable startup delay impact as has been reported with using `entry_points/console_script` (#7469).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by suominen** [comment]
I've tested it with
* Python 2.7.16 and 3.7.3 on Debian 10.1 (buster).
* Python 2.7.10 and 3.7.4 on macOS 10.14.6 (Mojave).

**Comment by suominen** [comment]
@fengzhou-msft - Didn't mean to dismiss any reviews, but just keep this branch without merge conflicts. Maybe it is better I don't resolve conflicts going forward -- it is just about `HISTORY.rst` changing in `dev`.

**Comment by yonzhan** [comment]
@fengzhou-msft  please help deal with this PR.

File: src/azure-cli/az [hunk 1]
```diff
@@ -1,14 +1,16 @@
-#!/usr/bin/env bash
+#!/usr/bin/env python
```
Comment (haroldrandom) [inline src/azure-cli/az]: Any earnings changing from bash to python?

Comment (suominen) [reply src/azure-cli/az]: I listed the benefits when opening this pull request: https://github.com/Azure/azure-cli/pull/10516#issue-316767937

Does that answer your question, or did you mean something else by "earnings"?

**Comment by fengzhou-msft** [comment]
@yonzhan I have tested the code, it works as stated and solved several issues. This change impacts the entry point for running all commands when installed with pip. We need @zikalino to review and see if there's any drawbacks.

**Comment by yonzhan** [comment]
@zikalino  please review this PR.

**Pull request merged**